File: src/main/java/no/novari/flyt/archive/gateway/resource/model/ResourceCollection.java
package no.novari.flyt.archive.gateway.resource.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.core.type.TypeReference;
import no.fint.model.resource.AbstractCollectionResources;

import java.util.List;

public class ResourceCollection extends AbstractCollectionResources<Object> {
    @Override
    @JsonIgnore
    @Deprecated
    public TypeReference<List<Object>> getTypeReference() {
        return new TypeReference<>() {
        };
    }

    public ResourceCollection() {
    }
}

File: src/main/kotlin/no/novari/flyt/archive/gateway/resource/model/ResourceCollection.kt
package no.novari.flyt.archive.gateway.resource.model

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.core.type.TypeReference
import no.novari.fint.model.resource.AbstractCollectionResources

class ResourceCollection : AbstractCollectionResources<Any>() {
    @JsonIgnore
    @Deprecated("Use collection type metadata from the concrete resource collection")
    override fun getTypeReference(): TypeReference<List<Any>> = object : TypeReference<List<Any>>() {}
}
